groupcache.go

```diff
@@ -272,21 +272,78 @@ func (g *Group) Set(ctx context.Context, key string, value []byte, expire time.T
 	}
 
 	_, err := g.setGroup.Do(key, func() (interface{}, error) {
+		wg := sync.WaitGroup{}
+		errs := make(chan error)
+
 		// If remote peer owns this key
 		owner, ok := g.peers.PickPeer(key)
 		if ok {
-			if err := g.setFromPeer(ctx, owner, key, value, expire); err != nil {
+			if err := g.setFromPeer(ctx, owner, key, value, expire, false); err != nil {
 				return nil, err
 			}
 			// TODO(thrawn01): Not sure if this is useful outside of tests...
 			//  maybe we should ALWAYS update the local cache?
-			if hotCache {
-				g.localSet(key, value, expire, &g.hotCache)
+			if !hotCache {
+				return nil, nil
+			}
+
+			g.localSet(key, value, expire, &g.hotCache)
+
+			for _, peer := range g.peers.GetAll() {
```

Comment:
I'd like to see this code deduplicated with its counterpart in the owner code path (line 324).  Move to function?